docs/java-7-rationale.md

```diff
@@ -33,8 +33,37 @@ But bytecode instrumentation tends to be much more sensitive to Java versions th
 we would lose a lot of confidence in the quality of our Java 7 support without being able to run our
 standard tests against it.
 
+Another option would be to run the "code under test" in a separate JVM from our test harness, which
```

Comment:
the option brave uses is retrolambda + animal sniffer until https://youtrack.jetbrains.com/issue/IDEA-85478

whether your tests execute in JRE 1.7 is less important than having code compile correctly. You can use a different process to smoke test a few paths as it is unlikely to break as long as you lock signatures and compilation levels.

Reply:
Instana has a test harness that forks test apps and runs test agents/instrumentations. The test harness uses testcontainers. It can compile and run the test apps in various JVM vendors, versions, languages (java, groovy, scala, kotlin..). There is an API that communicates between the app under test and the harness.

Its a lot of work. So in my opinion an honest answer to the "rationale" is stating the amount of work the project is willing to invest into these kind of compatibility tests. It also would be honest to state that the project does NOT do any kid of compatibility tests.